packages/mui-material-next/src/Menu/Menu.test.js

```diff
@@ -0,0 +1,793 @@
+      });
+    });
+  });
+
+  it('should pass `classes.paper` to the Paper', () => {
+    render(
+      <Menu
+        anchorEl={document.createElement('div')}
+        open
+        PaperProps={{ 'data-testid': 'paper' }}
+      />,
+    );
+
+    expect(screen.getByTestId('paper')).to.have.class(classes.paper);
+  });
+
+  describe('prop: PopoverClasses', () => {
+    it('should be able to change the Popover style', () => {
+      render(
+        <Menu
+          anchorEl={document.createElement('div')}
+          open
+          PaperProps={{ 'data-testid': 'paper' }}
+          PopoverClasses={{ paper: 'bar' }}
+        />,
+      );
+
+      expect(screen.getByTestId('paper')).to.have.class('bar');
+    });
+
+    it('should be able to change the Popover root element style when Menu classes prop is also provided', () => {
+      render(
+        <Menu
+          anchorEl={document.createElement('div')}
+          open
+          data-testid="popover"
+          classes={{ paper: 'bar' }}
+          PopoverClasses={{ root: 'foo' }}
+        />,
+      );
+      expect(screen.getByTestId('popover')).to.have.class('foo');
+    });
+  });
+
+  describe('prop: PaperProps', () => {
+    it('should be passed to the paper component', () => {
+      const customElevation = 12;
+      const customClasses = { rounded: { borderRadius: 12 } };
+      const wrapper = mount(
+        <Menu
+          anchorEl={document.createElement('div')}
+          open
+          PaperProps={{
+            'data-testid': 'paper',
+            elevation: customElevation,
+            classes: customClasses,
+          }}
+        />,
+      );
+
+      expect(wrapper.find(MenuPaper).props().elevation).to.equal(customElevation);
+      expect(wrapper.find(MenuPaper).props().classes).to.contain(customClasses);
+    });
+  });
+
+  it('should pass onClose prop to Popover', () => {
+    const handleClose = spy();
+    render(<Menu anchorEl={document.createElement('div')} open onClose={handleClose} />);
+
+    act(() => {
+      screen.getByRole('menu').focus();
+    });
+    fireEvent.keyDown(screen.getByRole('menu'), { key: 'Escape' });
+
+    expect(handleClose.callCount).to.equal(1);
+  });
+
+  it('renders its children only when open', () => {
+    const { setProps } = render(
+      <Menu anchorEl={document.createElement('div')} open={false}>
+        <div data-testid="children" />
+      </Menu>,
+    );
+
+    expect(screen.queryByTestId('children')).to.equal(null);
+
+    setProps({ open: true });
+
+    expect(screen.getByTestId('children')).not.to.equal(null);
+  });
+
+  describe('list node', () => {
+    it('should render a menu inside the Popover', () => {
+      render(<Menu anchorEl={document.createElement('div')} open data-testid="popover" />);
+
+      expect(screen.getByTestId('popover').querySelector('[role="menu"]')).not.to.equal(null);
+    });
+  });
+
+  // TODO: This test is failing
+  // it('should open during the initial mount', () => {
```

Comment:
@michaldudak this test is failing for some reason. From what I could see, the difference is that this MenuItem is not using the useMenuItem hook. I am not sure if this is the cause for the fail tough.

Reply:
At first glance it seems that the MenuItem doesn't get the `autoFocus` prop from anywhere.